host_core/lib/host_core/linkdefs/manager.ex

```diff
@@ -34,19 +38,20 @@ defmodule HostCore.Linkdefs.Manager do
 
   def put_link_definition(actor, contract_id, link_name, provider_key, values) do
     ldid = UUID.uuid4()
-    cache_link_definition(ldid, actor, contract_id, link_name, provider_key, values)
 
-    ld = %{
-      id: ldid,
-      actor_id: actor,
-      provider_id: provider_key,
-      link_name: link_name,
-      contract_id: contract_id,
-      values: values,
-      deleted: false
-    }
+    with :ok <- cache_link_definition(ldid, actor, contract_id, link_name, provider_key, values) do
+      ld = %{
+        id: ldid,
+        actor_id: actor,
+        provider_id: provider_key,
+        link_name: link_name,
+        contract_id: contract_id,
+        values: values,
+        deleted: false
+      }
 
-    publish_link_definition(ld)
+      publish_link_definition(ld)
+    end
```

Comment:
TIL this returns the value that doesn't match `:ok` as a default with no `else` required